x/stake/keeper/validator.go

```diff
@@ -43,13 +38,13 @@ func (k Keeper) GetValidator(ctx sdk.Context, addr sdk.ValAddress) (validator ty
 	// amino bytes weren't found in cache, so amino unmarshal and add it to the cache
 	validator = types.MustUnmarshalValidator(k.cdc, addr, value)
 	cachedVal := cachedValidator{validator, strValue}
-	validatorCache[strValue] = cachedValidator{validator, strValue}
-	validatorCacheList.PushBack(cachedVal)
+	k.validatorCache[strValue] = cachedValidator{validator, strValue}
+	k.validatorCacheList.PushBack(cachedVal)
 
 	// if the cache is too big, pop off the last element from it
-	if validatorCacheList.Len() > 500 {
-		valToRemove := validatorCacheList.Remove(validatorCacheList.Front()).(cachedValidator)
-		delete(validatorCache, valToRemove.marshalled)
+	if k.validatorCacheList.Len() > 500 {
```

Comment:
The 500 here and in the constructor should honestly be a constant imho.

Reply:
So moved!